src/main/java/com/appland/appmap/record/RecordingSessionGeneric.java

```diff
@@ -2,44 +2,57 @@
 
 import com.appland.appmap.output.v1.CodeObject;
 import com.appland.appmap.output.v1.Event;
-import com.appland.appmap.util.Logger;
 
-import java.util.Vector;
 import java.util.HashSet;
+import java.util.Optional;
+import java.util.Vector;
 
 public class RecordingSessionGeneric implements IRecordingSession {
-  protected Vector<Event> events = new Vector<Event>();
-  protected HashSet<String> classReferences = new HashSet<>();
-
-  public synchronized void add(Event event) {
-    this.events.add(event);
-
-    final String key = event.definedClass +
-      ":" + event.methodId +
-      ":" + event.isStatic + 
-      ":" + event.lineNumber;
-    this.classReferences.add(key);
-  }
-
-  public void start() {
-    throw new UnsupportedOperationException();
-  }
-
-  public String stop() {
-    throw new UnsupportedOperationException();
-  }
-
-  protected CodeObjectTree getClassMap() {
-    CodeObjectTree registeredObjects = Recorder.getInstance().getRegisteredObjects();
-    CodeObjectTree classMap = new CodeObjectTree();
-    for (String key: this.classReferences) {
-      String[] parts = key.split(":");
-     
-      CodeObject methodBranch = registeredObjects.getMethodBranch(parts[0], parts[1], Boolean.valueOf(parts[2]), Integer.valueOf(parts[3]));
-      if (methodBranch != null)
-        classMap.add(methodBranch);
+    protected Vector<Event> events = new Vector<Event>();
+    protected HashSet<String> classReferences = new HashSet<>();
+
+    public synchronized void add(Event event) {
+        this.events.add(event);
+
+        if (event.parentId == null) {
+            this.classReferences.add(
+                    event.definedClass +
+                            ":" + event.methodId +
+                            ":" + event.isStatic +
+                            ":" + event.lineNumber);
+        } else {
+          Optional.ofNullable(events.stream()
```

Comment:
@dustinbyrne should we keep things below ava 8 ?

Reply:
This is fine. We're using other Java 8 features elsewhere.